File: MiniAirbnbBackend/src/main/java/com/airbnb/miniairbnb/service/UserService.java
package com.airbnb.miniairbnb.service;

import com.airbnb.miniairbnb.model.User;
import com.airbnb.miniairbnb.model.UserRole;

import java.util.List;
import java.util.Optional;

public interface UserService {
    User registerUser(String email, String password, String firstName, String lastName, UserRole role); //inregistrare utilizator nou

    Optional<User> findUserByEmail(String email); //gaseste utilizatorul dupa email

    Optional<User> findUserById(Long id); //gaseste utilizator dupa id

    List<User> findUsersByRole(UserRole role); //gaseste toti utilizatorii dupa rol

    boolean emailExists(String email); //verifica daca un email exista deja

    List<User> findAllUsers(); //gaseste toti utilizatorii, doar adminul

    User updateUserRole(Long userId, UserRole newRole);

    void deleteUser(Long userId);
}

File: mini-airbnb-frontend/src/services/userService.ts
import api from "./api";
import type { User, UserRole } from "../types";

export const userService = {
  // Obține toți utilizatorii (doar Admin)
  getAllUsers: async (): Promise<User[]> => {
    const response = await api.get<User[]>("/users");
    return response.data;
  },

  // Schimbă rolul unui utilizator (doar Admin)
  updateUserRole: async (userId: number, role: UserRole): Promise<User> => {
    const response = await api.put<User>(`/users/${userId}/role`, { role });
    return response.data;
  },

  // Șterge un utilizator (doar Admin)
  deleteUser: async (userId: number): Promise<void> => {
    await api.delete(`/users/${userId}`);
  },
};

